numba/typed/typedlist.py

```diff
@@ -237,7 +237,9 @@ def _typed(self):
 
     @property
     def _dtype(self):
-        return self._list_type.dtype if self._typed else None
+        if not self._typed:
+            raise TypeError("invalid operation on untyped list")
```

Comment:
I'm unconvinced that this is a `TypeError`, what's being applied to an inappropriate type and what is the inappropriate type? Perhaps `RuntimeError`?

Reply:
O.K. I made the fix, should the `_numba_type_` property consequently also raise a `RuntimeError`rather than a `TypeError`?

Reply:
Thanks, I think probably yes. Unless I'm mistaken, the issue here is not type related?

Reply:
O.K. I think I will make a separate PR for this, perhaps. Typed-dict suffers from this also, btw:

https://github.com/numba/numba/blob/master/numba/typed/typeddict.py#L134

So, maybe merge this one now and then fixing the incorrect `TypeError` for both dict and list in it's own PR?

Reply:
Sounds good. Thanks.